cli/cmd/plugin/unmanaged-cluster/cmd/docs.go

```diff
@@ -0,0 +1,170 @@
+	docsOutputDir = ""
+	noStdout      = false
+)
+
+// ListCmd returns a list of existing clusters.
+var DocsCmd = &cobra.Command{
+	Use:   "docs",
+	Short: "Generated documentation & usage of unmanaged-cluster",
+	Long:  docsDesc,
+	PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+		return nil
+	},
+	RunE: startDocsGen,
+	PostRunE: func(cmd *cobra.Command, args []string) (err error) {
+		return nil
+	},
+}
+
+func init() {
+	DocsCmd.Flags().StringVar(&docsFormat, "format", "markdown", "Output format (markdown|manpage|rest|yaml); default is markdown")
+	DocsCmd.Flags().BoolVar(&noStdout, "no-stdout", false, "Ignores printing to stdout and inspects 'output-dir' for location to dump files")
+	DocsCmd.Flags().StringVar(&docsOutputDir, "output-dir", "", "Location to output the doc page files; providing this prevents printing to stdout. Defaults to the current working dir")
+}
+
+// list outputs a list of all unmanaged clusters on the system.
+func startDocsGen(cmd *cobra.Command, args []string) error {
+	var err error
+
+	if len(args) != 0 {
+		return fmt.Errorf("docs generation does not support any arguments")
+	}
+
+	// Valid formats
+	if (docsFormat != "markdown" && docsFormat != "manpage") && (docsFormat != "rest" && docsFormat != "yaml") {
+		return fmt.Errorf("unsupported format: %s - Valid formats: (markdown|manpage|rest|yaml)", docsFormat)
+	}
+
+	if noStdout || cmd.Flags().Changed("output-dir") {
+		err = docsToFiles(cmd)
+	} else {
+		err = docsToStdout(cmd)
+	}
+
+	return err
+}
+
+func docsToFiles(cmd *cobra.Command) error {
+	if docsOutputDir == "" {
+		p, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		docsOutputDir = p
+	}
+
+	switch docsFormat {
+	case "markdown":
+		err := doc.GenMarkdownTree(cmd.Parent().Root(), docsOutputDir)
+		if err != nil {
+			return err
+		}
+	case "manpage":
+		manHeader := &doc.GenManHeader{
+			Title:   "Tanzu Community Edition",
+			Section: "1",
+		}
+		err := doc.GenManTree(cmd.Parent().Root(), manHeader, docsOutputDir)
+		if err != nil {
+			return err
+		}
+	case "rest":
+		err := doc.GenReSTTree(cmd.Parent().Root(), docsOutputDir)
+		if err != nil {
+			return err
+		}
+	case "yaml":
+		err := doc.GenYamlTree(cmd.Parent().Root(), docsOutputDir)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+func docsToStdout(cmd *cobra.Command) error {
+	buf := new(bytes.Buffer)
+
+	switch docsFormat {
+	case "markdown":
+		if err := doc.GenMarkdown(cmd.Root(), buf); err != nil {
+			return err
+		}
+
+		for _, c := range cmd.Root().Commands() {
+			if err := doc.GenMarkdown(c, buf); err != nil {
+				return err
+			}
+		}
+	case "manpage":
+		manHeader := &doc.GenManHeader{
+			Title:   "Tanzu Community Edition",
+			Section: "1",
+		}
+
+		if err := doc.GenMan(cmd.Root(), manHeader, buf); err != nil {
+			return err
+		}
+
+		for _, c := range cmd.Root().Commands() {
+			if err := doc.GenMan(c, manHeader, buf); err != nil {
+				return err
+			}
+		}
+	case "rest":
```

Comment:
"rest" probably isn't the right term to use for reStructuredText. I would keep it "rst"

Reply:
I was mostly going off what `cobra` has documented: https://github.com/spf13/cobra/blob/master/doc/rest_docs.md

Looks like it's usually referred to as `ReST` which seemed wonky as a config option

Reply:
Looks like on the official spec it gets refferred to as either `reStructuredText` or `ReST`: https://docutils.sourceforge.io/rst.html

Reply:
Huh, I've spent more time with RST having been in Python land for years, and I've never seen it referred to "rest". I would take that to mean some sort of API. :D 

If that's how they have it documented, then I guess that's good enough for me!